base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -1086,6 +1159,350 @@ where
+        let _single_round_sender_data = stp
+            .build_single_round_message(&self.resources.transaction_key_manager_service)
+            .await
+            .map_err(|e| TransactionServiceProtocolError::new(tx_id, e.into()))?;
+
+        self.resources
+            .output_manager_service
+            .confirm_pending_transaction(tx_id)
+            .await
+            .map_err(|e| TransactionServiceProtocolError::new(tx_id, e.into()))?;
+
+        // Prepare receiver part of the transaction
+
+        // In generating an aggregate public key utxo, we can use a randomly generated spend key
+        let spending_key = PrivateKey::random(&mut OsRng);
+        let encryption_private_key = secret_key_to_output_encryption_key(&spending_key)?;
+
+        let sender_offset_private_key = stp
+            .get_recipient_sender_offset_private_key()
+            .map_err(|e| TransactionServiceProtocolError::new(tx_id, e.into()))?
+            .ok_or(TransactionServiceProtocolError::new(
+                tx_id,
+                TransactionServiceError::InvalidKeyId("Missing sender offset keyid".to_string()),
+            ))?;
+
+        let encryption_key_id = self
+            .resources
+            .transaction_key_manager_service
+            .import_key(encryption_private_key)
+            .await?;
+
+        let sender_offset_public_key = self
+            .resources
+            .transaction_key_manager_service
+            .get_public_key_at_key_id(&sender_offset_private_key)
+            .await?;
+
+        let spending_key_id = self
+            .resources
+            .transaction_key_manager_service
+            .import_key(spending_key.clone())
+            .await?;
+
+        let minimum_value_promise = MicroMinotari::zero();
+        let covenant = Covenant::default();
+        let wallet_output = WalletOutputBuilder::new(amount, spending_key_id)
+            .with_features(
+                sender_message
+                    .single()
+                    .ok_or(TransactionServiceProtocolError::new(
+                        tx_id,
+                        TransactionServiceError::InvalidMessageError("Sent invalid message type".to_string()),
+                    ))?
+                    .features
+                    .clone(),
+            )
+            .with_script(script)
+            // We don't want the given utxo to be spendable as an input to a later transaction, so we set
+            // spendable height of the current utxo to be u64::MAx
+            .with_script_lock_height(u64::MAX)
+            .encrypt_data_for_recovery(
+                &self.resources.transaction_key_manager_service,
+                Some(&encryption_key_id),
+                PaymentId::Empty,
+            )
+            .await?
+            .with_input_data(
+                // TODO: refactor this, when we have implemented the necessary logic
+                inputs!(PublicKey::from_secret_key(&spending_key)),
```

Comment:
I dont think we can add anything here, this will be the input stack you need to spend. This will only be available at spend